core/templates/pages/blog-author-profile-page/blog-author-profile-page-root.component.spec.ts

```diff
@@ -0,0 +1,254 @@
+      .not.toHaveBeenCalledWith();
+    expect(component.pageIsShown).toBeFalse();
+    expect(component.errorPageIsShown).toBeTrue();
+    expect(loaderService.hideLoadingScreen).toHaveBeenCalled();
+  }));
+
+  it('should initialize and validate access when blog project feature is ' +
+  'disabled but user can edit blog posts', fakeAsync(() => {
+    mockPlatformFeatureService.status.BlogProject.isEnabled = false;
+    spyOn(userService, 'canUserEditBlogPosts').and.returnValue(
+      Promise.resolve(true));
+    spyOn(
+      accessValidationBackendApiService,
+      'validateAccessToBlogAuthorProfilePage'
+    ).and.returnValue(Promise.resolve());
+    spyOn(loaderService, 'showLoadingScreen');
+    spyOn(loaderService, 'hideLoadingScreen');
+
+    component.ngOnInit();
+    tick();
+
+    expect(loaderService.showLoadingScreen).toHaveBeenCalled();
+    expect(
+      accessValidationBackendApiService.validateAccessToBlogAuthorProfilePage)
+      .toHaveBeenCalledWith('author');
+    expect(component.errorPageIsShown).toBeFalse();
+    expect(loaderService.hideLoadingScreen).toHaveBeenCalled();
+  }));
+
+  it('should initialize and subscribe to onLangChange', fakeAsync(() => {
+    spyOn(
+      accessValidationBackendApiService,
+      'validateAccessToBlogAuthorProfilePage'
+    ).and.returnValue(Promise.resolve());
+    spyOn(component.directiveSubscriptions, 'add');
+    spyOn(translateService.onLangChange, 'subscribe');
+
+    component.ngOnInit();
+    tick();
+
+    expect(component.directiveSubscriptions.add).toHaveBeenCalled();
+    expect(translateService.onLangChange.subscribe).toHaveBeenCalled();
+  }));
+
+  it('should update page title whenever the language changes', () => {
+    spyOn(
+      accessValidationBackendApiService,
+      'validateAccessToBlogAuthorProfilePage'
+    ).and.returnValue(Promise.resolve());
+    component.ngOnInit();
+    spyOn(component, 'setPageTitleAndMetaTags');
+
+    translateService.onLangChange.emit();
+
+    expect(component.setPageTitleAndMetaTags).toHaveBeenCalled();
+  });
+
+  it('should obtain translated title and set the title and meta tags', () => {
+    spyOn(translateService, 'instant').and.callThrough();
+    spyOn(pageHeadService, 'updateTitleAndMetaTags');
+
+    component.setPageTitleAndMetaTags();
+
+    expect(translateService.instant).toHaveBeenCalledWith(
+      AppConstants.PAGES_REGISTERED_WITH_FRONTEND.BLOG_AUTHOR_PROFILE_PAGE.TITLE
+    );
+    expect(pageHeadService.updateTitleAndMetaTags).toHaveBeenCalledWith(
+      AppConstants.PAGES_REGISTERED_WITH_FRONTEND
+        .BLOG_AUTHOR_PROFILE_PAGE.TITLE,
+      AppConstants.PAGES_REGISTERED_WITH_FRONTEND.BLOG_AUTHOR_PROFILE_PAGE.META
+    );
+  });
+
+  it('should unsubscribe on component destruction', () => {
+    spyOn(component.directiveSubscriptions, 'unsubscribe');
+
+    component.ngOnDestroy();
+
+    expect(component.directiveSubscriptions.unsubscribe).toHaveBeenCalled();
```

Comment:
Add `expect(component.directiveSubscriptions.closed).toBe(true);` to ensure no subscriptions are left.

Reply:
Done